lib/MilkCheck/Engine/BaseEntity.py

```diff
@@ -421,13 +421,13 @@ def remove_dep(self, dep_name, parent=True):
 
     def clear_parent_deps(self):
         '''Remove all parent dependencies of an entity'''
-        for dpname in self.parents.keys():
-            self.remove_dep(dpname)
+        dpnames = list(self.parents.keys())
```

Comment:
I mean keeping the original for-loop, just adding a `list()` call.

        for dpname in list(self.parents.keys()):
            self.remove_dep(dpname)

same below :)

Reply:
You're right ;)
Should be ok now.